agent-inject/agent/agent.go

```diff
@@ -335,11 +343,18 @@ func New(pod *corev1.Pod, patches []*jsonpatch.JsonPatchOperation) (*Agent, erro
 		return agent, err
 	}
 
+	agentCacheExitOnErr, err := agent.agentCacheExitOnErr()
+	if err != nil {
+		return agent, err
+	}
+
 	agent.VaultAgentCache = VaultAgentCache{
 		Enable:           agentCacheEnable,
 		ListenerPort:     pod.Annotations[AnnotationAgentCacheListenerPort],
 		UseAutoAuthToken: pod.Annotations[AnnotationAgentCacheUseAutoAuthToken],
+		ExitOnErr:        agentCacheExitOnErr,
 	}
+	agent.VaultAgentCache.Persist = agent.agentCachePersist()
```

Comment:
```suggestion
		ExitOnErr:        agentCacheExitOnErr,
		Persist:          agent.agentCachePersist(),
	}
```

Reply:
Yep, I reworked this a bit in 00f91529016d93f037dde237a0897e98500cab96 to set it all in the struct, since `agentCachePersist()` also checks whether the cache is enabled.